src/Application/PlexAccounts/CheckUsername/CheckUsernameTaskHandler.cs

```diff
@@ -0,0 +1,67 @@
+using Application.Contracts;
+using Data.Contracts;
+using FluentValidation;
+using Logging.Interface;
+
+namespace PlexRipper.Application.PlexAccounts.CheckUsername;
+
+public class CheckUsernameTaskValidator : AbstractValidator<CheckUsernameTaskRequest>
+{
+    #region Constructors
+
+    public CheckUsernameTaskValidator()
+    {
+        RuleFor(x => x.Username).NotEmpty();
+        RuleFor(x => x.Username).MinimumLength(5);
+    }
+
+    #endregion
+}
+
+public class CheckUsernameTaskHandler : IRequestHandler<CheckUsernameTaskRequest, Result<bool>>
+{
+    #region Fields
+
+    private readonly ILog _log;
+    private readonly IMediator _mediator;
+
+    #endregion
+
+    #region Constructors
+
+    public CheckUsernameTaskHandler(ILog log, IMediator mediator)
+    {
+        _log = log;
+        _mediator = mediator;
+    }
+
+    #endregion
+
+    #region Methods
+
+    #region Public
+
+    public async Task<Result<bool>> Handle(CheckUsernameTaskRequest request, CancellationToken cancellationToken)
+    {
+        var result = await _mediator.Send(new GetPlexAccountByUsernameQuery(request.Username), cancellationToken);
```

Comment:
Remember when we talked about "streamlining" your flows? 
I wouldn't expect the `CheckUsernameTaskRequest` to send another IRequest somewhere. Instead, I'd expect a repository or a service that would query a storage (or something else).

Reply:
Yes and this `GetPlexAccountByUsernameQuery` is a database access query/handler situated in the `Data` project which only contains EF code. I figured this was acceptable because I don't go more than 2 handlers deep and the Data project is part of the Infrastructure layer.

Reply:
Alright, this is where we have to make compromises. Given you have a single project for you API, you can't dump a dependency to EF in there; at least, this is what fancier architectures allow you to do.
Two approaches are possible:
- Change of plans: you `Data` folder only contains the core of EF -> Models and Context, and you're allowed to use EF features (IDbContext) in an API, creating a dependency. This way, this code could move in the same folder.
- You don't want a dependency to EF in your API project. In that case, I'd just recommend not to use mediator because it makes things harder to understand. Note that the content of Data will be harder to maintain, as it will contain interactions for all use cases